ansible_collections/arista/avd/roles/eos_designs/python_modules/base/snmp_server.py

```diff
@@ -0,0 +1,244 @@
+            # Empty list or None
+            return None
+
+        snmp_hosts = []
+
+        has_mgmt_ip = (self.shared_utils.mgmt_ip is not None) or (self.shared_utils.ipv6_mgmt_ip is not None)
+
+        for host in natural_sort(hosts, "host"):
+            # Initialize a set with vrf defined under the host (Catching None value with or [])
+            vrfs = set()
+            if (vrf := host.pop("vrf", None)) is not None:
+                vrfs.add(vrf)
+            if (use_mgmt_interface_vrf := host.pop("use_mgmt_interface_vrf", False)) is True and has_mgmt_ip:
+                vrfs.add(self.shared_utils.mgmt_interface_vrf)
+
+            if (use_inband_mgmt_vrf := host.pop("use_inband_mgmt_vrf", False)) is True and self.shared_utils.inband_mgmt_interface is not None:
+                # self.shared_utils.inband_mgmt_vrf returns None for the default VRF, but here we need "default" to avoid duplicates.
+                vrfs.add(self.shared_utils.inband_mgmt_vrf or "default")
+
+            if not any([vrfs, use_mgmt_interface_vrf, use_inband_mgmt_vrf]):
+                # If no VRFs are defined (and we are not just ignoring missing mgmt config)
+                vrfs.add("default")
+
+            # Ensure default VRF is added first
+            if "default" in vrfs:
+                vrfs.remove("default")
+                # Add host without VRF field
+                snmp_hosts.append(host)
+
+            for vrf in natural_sort(vrfs):
+                # Add host with VRF field.
+                snmp_hosts.append({**host, "vrf": vrf})
+
+        return snmp_hosts or None
+
+    def _snmp_local_interfaces(self, source_interfaces_inputs: dict | None) -> list | None:
+        """
+        Return local_interfaces if "source_interfaces.snmp" is set.
+        Otherwise return None
+        """
+
+        if not source_interfaces_inputs:
+            # Empty dict or None
+            return None
+
+        local_interfaces = self._build_source_interfaces(
+            source_interfaces_inputs.get("mgmt_interface", False), source_interfaces_inputs.get("inband_mgmt_interface", False), "SNMP"
+        )
+        return local_interfaces or None
+
+    def _snmp_vrfs(self, snmp_settings: dict | None) -> list | None:
+        """
+        Return list of dicts for enabling/disabling SNMP for VRFs
+        Requires one of the following options to be set under snmp_settings:
+        - vrfs
+        - enable_mgmt_interface_vrf
+        - enable_inband_mgmt_vrf
+        Otherwise return None
+        """
+        if snmp_settings is None:
+            return None
+
+        has_mgmt_ip = (self.shared_utils.mgmt_ip is not None) or (self.shared_utils.ipv6_mgmt_ip is not None)
+
+        # Initialize a set with all vrfs (Catching None value with or [])
+        vrfs = get(snmp_settings, "vrfs", default=[])
+        if (enable_mgmt_interface_vrf := snmp_settings.get("enable_mgmt_interface_vrf")) is not None and has_mgmt_ip:
```

Comment:
```suggestion
        if has_mgmt_ip and (enable_mgmt_interface_vrf := snmp_settings.get("enable_mgmt_interface_vrf")) is not None:
```
If the has_mgmt_ip is False then the second entity will not be invoked. So changing the order

Reply:
done